internal/bminventory/inventory.go

```
@@ -81,20 +86,76 @@ type debugCmd struct {
 	stepID string
 }
 
+type Command struct {
+	Name                string
+	Command             string
+	Arguments           []string
+	ExpectedArgumentNum int
+}
+
+type AgentCommands struct {
+	Commands []Command
+}
+
 type bareMetalInventory struct {
 	Config
 	imageBuildCmd []string
 	db            *gorm.DB
 	kube          client.Client
 	debugCmdMap   map[strfmt.UUID]debugCmd
 	debugCmdMux   sync.Mutex
+	agentCommands AgentCommands
+}
+
+func (b *bareMetalInventory) readAgentCommands() error {
+	contents, err := ioutil.ReadFile(AgentsCommadsFile)
+	if err != nil {
+		logrus.WithError(err).Errorf("Could not read file %s", AgentsCommadsFile)
+		return err
+	}
+	err = yaml.Unmarshal(contents, &b.agentCommands)
+	if err != nil {
+		logrus.WithError(err).Errorf("Error unmarshaling %s", contents)
+	}
+	return nil
+}
+
+func (b *bareMetalInventory) getCommandByName(name string) *Command {
+	for _, cmd := range b.agentCommands.Commands {
+		if cmd.Name == name {
+			return &cmd
+		}
+	}
+	logrus.Warnf("Command %s was not found", name)
+	return nil
+}
+
+func (b *bareMetalInventory) createAgentExecuteStep(cmdName string, args ...string) *models.Step {
+	cmd := b.getCommandByName(cmdName)
+	if cmd == nil {
+		return nil
```

return error

---

Error is useless here since the error was already logged

---

i think that @ronniel1  is not talking about the log. We tried to execute a command that doesn't exists. the api should return an error.